File: newsmodule/src/main/java/zjm/cst/dhu/newsmodule/mvp/view/NewsListContract.java
package zjm.cst.dhu.newsmodule.mvp.view;

import java.util.List;

import zjm.cst.dhu.library.mvp.model.NewsChannel;
import zjm.cst.dhu.library.mvp.model.NewsData;
import zjm.cst.dhu.library.mvp.presenter.BasePresenter;
import zjm.cst.dhu.library.mvp.view.base.BaseView;

/**
 * Created by zjm on 2017/5/10.
 */

public interface NewsListContract {
    interface View extends BaseView {
        void loadNewListSuccess(List<NewsData> list);

        void loadNewListError();

        void loadNextListSuccess(List<NewsData> list);

        void loadNextListError();

        void floatingActionButtonClick();
    }

    interface Presenter extends BasePresenter<View> {
        void setValue(String id,String type);

        void loadNewList();

        void loadNextList();
    }
}

File: newsmodule/src/main/kotlin/zjm.cst.dhu.newsmodule/mvp/view/NewsListContract.kt
package zjm.cst.dhu.newsmodule.mvp.view


import zjm.cst.dhu.library.mvp.model.NewsData
import zjm.cst.dhu.library.mvp.presenter.BasePresenter
import zjm.cst.dhu.library.mvp.view.base.BaseView

/**
 * Created by zjm on 2017/5/10.
 */

interface NewsListContract {
    interface View
        : BaseView {
        fun loadNewListSuccess(list: List<NewsData>)

        fun loadNewListError()

        fun loadNextListSuccess(list: List<NewsData>)

        fun loadNextListError()

        fun floatingActionButtonClick()
    }

    interface Presenter
        : BasePresenter<View> {

        fun setValue(id: String, type: String)

        fun loadNewList()

        fun loadNextList()
    }
}
